src/qtism/data/storage/xml/marshalling/ValueMarshaller.php

```diff
@@ -164,4 +167,30 @@ public function getExpectedQtiClassName()
     {
         return 'value';
     }
+
+    /**
+     * @param mixed $value
+     * @return DOMCdataSection|false
+     */
+    private function encloseValueInCDATASection($value)
+    {
+        return self::getDOMCradle()->createCDATASection(XmlUtils::valueAsString($value));
+    }
+
+    /**
+     * @param DOMElement $element
+     * @return string
+     */
+    private function extractValueFromCDATASection(DOMElement $element)
+    {
+        $valueChildNodes = self::getChildElements($element, true);
+        if (!empty($valueChildNodes)
+            && ($node = reset($valueChildNodes))
+            && $node->nodeType === XML_CDATA_SECTION_NODE
+        ) {
+            return trim($node->nodeValue);
+        } else {
+            return trim($element->nodeValue);
+        }
+    }
```

Comment:
`reset($valueChildNodes)` cannot be empty or undefined if `!empty($valueChildNodes)`. Besides `$node->nodeType === XML_CDATA_SECTION_NODE` should always be `true`. Otherwise, something went very wrong.
Also, let's separate the concerns just a tiny bit. WDYT of the following?

```suggestion
        $valueNode = $this->extractValueNode($element);

        return trim($valueNode->nodeValue);
    }

    private function extractValueNode(DOMElement $element): DOMElement
    {
        $valueChildNodes = self::getChildElements($element, true);

        foreach(self::getChildElements($element, true) as $valueNode) {
            if ($valueNode->nodeType === XML_CDATA_SECTION_NODE) {
                return $valueNode;
            }
        }

        return $element;
    }
```

Reply:
> reset($valueChildNodes) cannot be empty or undefined if !empty($valueChildNodes)

Yes, that is what I expect, I use it to pick the first found node and create a new variable at the same time. 

> Besides $node->nodeType === XML_CDATA_SECTION_NODE should always be true.

No, not necessarily. Since loaded XML values might not have the CDATA wrapping, to make it a non-breaking change I suggest for `<value>` element to accept text or CDATA wrapped text.

Reply:
> No, not necessarily.

But it is, when taken from `self::getChildElements()`.

> to make it a non-breaking change I suggest for <value> element to accept text or CDATA wrapped text

Yeah, that's clear :)
I just suggest to separate the value-containing node extraction logic from the value extraction itself.

Reply:
I tested that returned child nodes would return both `DOMText` and `DOMCdataSection` nodes, so the code could be simplified